pkg/controller/podreadyupdater/podreadyupdater_controller.go

```diff
@@ -0,0 +1,311 @@
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	appconfig "github.com/openyurtio/openyurt/cmd/yurt-manager/app/config"
+	appsv1beta1 "github.com/openyurtio/openyurt/pkg/apis/apps/v1beta1"
+	nodeutil "github.com/openyurtio/openyurt/pkg/controller/util/node"
+	podutil "github.com/openyurtio/openyurt/pkg/controller/util/pod"
+	"github.com/openyurtio/openyurt/pkg/controller/yurtcoordinator/constant"
+	utilclient "github.com/openyurtio/openyurt/pkg/util/client"
+	utildiscovery "github.com/openyurtio/openyurt/pkg/util/discovery"
+)
+
+func init() {
+	flag.IntVar(&concurrentReconciles, "podreadyupdater-workers", concurrentReconciles, "Max concurrent workers for PodReadyUpdater controller.")
+}
+
+var (
+	concurrentReconciles = 3
+	controllerKind       = appsv1.SchemeGroupVersion.WithKind("Pod")
+)
+
+const (
+	ControllerName       = "podReadyUpdater"
+	labelCurrentNodePool = "apps.openyurt.io/nodepool"
+)
+
+func Format(format string, args ...interface{}) string {
+	s := fmt.Sprintf(format, args...)
+	return fmt.Sprintf("%s: %s", ControllerName, s)
+}
+
+// Add creates a new PodReadyUpdater Controller and adds it to the Manager with default RBAC. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func Add(c *appconfig.CompletedConfig, mgr manager.Manager) error {
+	if !utildiscovery.DiscoverGVK(controllerKind) {
+		return nil
+	}
+	return add(mgr, newReconciler(c, mgr))
+}
+
+var _ reconcile.Reconciler = &ReconcilePodReadyUpdater{}
+
+// ReconcilePodReadyUpdater reconciles a PodReadyUpdater object
+type ReconcilePodReadyUpdater struct {
+	client.Client
+	scheme *runtime.Scheme
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(c *appconfig.CompletedConfig, mgr manager.Manager) reconcile.Reconciler {
+	return &ReconcilePodReadyUpdater{
+		Client: utilclient.NewClientFromManager(mgr, ControllerName),
+		scheme: mgr.GetScheme(),
+	}
+}
+
+// add adds a new Controller to mgr with r as the reconcile.Reconciler
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	c, err := controller.New(ControllerName, mgr, controller.Options{
+		Reconciler: r, MaxConcurrentReconciles: concurrentReconciles,
+	})
+	if err != nil {
+		return err
+	}
+
+	// 1. Watch for changes to Pod
+	podReadyPredicate := predicate.Funcs{
+		CreateFunc: func(evt event.CreateEvent) bool {
+			return false
```

Comment:
we should also handle `Pod Create` event, because of `add events` will be received when yurt-manager restart.